### ISSUE TITLE
[306] set -o pipefail not detected unless it is the first line of the shell command

### ISSUE THREAD

**Issue by SpamapS** [ansible-community/ansible-lint#663]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
ansible-lint --version
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

I have some shell tasks that start like this:
set -e
set -o pipefail

This is more readable than

set -eo pipefail

I expect this to properly detect that I've used pipefail.

# Actual Behaviour (Bug report only)

This shell task is triggering 306.

We should split the task on the detected pipe, and only care about lines before the usage of pipes.
